internal/validation/apiextensions/v1/composition/handler.go

```diff
@@ -50,22 +44,21 @@ const (
 	crdsIndexKey = "crd.kind.group"
 )
 
-// handler implements the admission handler for Composition.
-type handler struct {
-	reader  client.Reader
-	decoder *admission.Decoder
-	options controller.Options
-}
+// Error strings.
+const (
+	errNotComposition = "supplied object was not a Composition"
+	errUnexpectedOp   = "unexpected operation"
+	errValidationMode = "cannot get validation mode"
 
-// InjectDecoder injects the decoder.
-func (h *handler) InjectDecoder(decoder *admission.Decoder) error {
-	h.decoder = decoder
-	return nil
-}
+	errFmtTooManyCRDs = "more than one CRD found for %s.%s: %v"
+	errFmtGetCRDs     = "cannot get the needed CRDs: %v"
+)
 
 // SetupWebhookWithManager sets up the webhook with the manager.
 func SetupWebhookWithManager(mgr ctrl.Manager, options controller.Options) error {
 	if options.Features.Enabled(features.EnableAlphaCompositionWebhookSchemaValidation) {
+		// TODO(negz): It's not obvious what this does and why. Someone who
+		// understands it should add a comment. :)
```

Comment:
@phisco Do you know. I can kinda guess but it was confusing enough that a comment would be useful.

Reply:
Something like this would make sense?
```suggestion
		// set up indexes to be able to list CRDs by Group and Kind they define later on
```